### PR TITLE
[integration test] react test function

### PR DESCRIPTION
**Pull request by kongtiaowang**
Make it easy to write UI react test.
Complete skip test functions.
Rewrite react test.

### CODE REVIEW

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 1]
```diff
@@ -37,7 +37,7 @@ abstract class LorisIntegrationTest extends TestCase
     protected $factory;
 
     private $_oldConfig = array();
-
+    private $NUM_OF_ATTEMPTS = 5;
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
    /** The maximum number of attempts the integration tests will try to execute some task before aborting.
    * @var int
    */
    private const MAX_ATTEMPTS = 5;
```

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 2]
```diff
@@ -535,5 +535,133 @@ abstract class LorisIntegrationTest extends TestCase
...
+                    "return document.querySelector('$ui').textContent"
+                );
+            } catch (Exception $e) {
+                echo 'Caught exception: ',  $e->getMessage(), "\n";
+                        $attempts++;
+                        sleep(1);
+                        continue;
+            }
+              break;
+        } while($attempts < $this->NUM_OF_ATTEMPTS);
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
        } while($attempts < self::MAX_ATTEMPTS);
```

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 3]
```diff
@@ -535,5 +535,133 @@ abstract class LorisIntegrationTest extends TestCase
...
+                    "document.querySelector('$ui').click()"
+                );
+            } catch (Exception $e) {
+                echo 'Caught exception: ',  $e->getMessage(), "\n";
+                        $attempts++;
+                        sleep(1);
+                        continue;
+            }
+            break;
+        } while($attempts < $this->NUM_OF_ATTEMPTS);
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
        } while($attempts < self::MAX_ATTEMPTS);
```

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 4]
```diff
@@ -535,5 +535,133 @@ abstract class LorisIntegrationTest extends TestCase
...
+                    "return document.querySelector('$ui').textContent"
+                );
+            } catch (Exception $e) {
+                echo 'Caught exception: ',  $e->getMessage(), "\n";
+                        $attempts++;
+                        sleep(1);
+                        continue;
+            }
+              break;
+        } while($attempts < $this->NUM_OF_ATTEMPTS);
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
        } while($attempts < self::MAX_ATTEMPTS);
```

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 5]
```diff
@@ -535,5 +535,133 @@ abstract class LorisIntegrationTest extends TestCase
...
+                 input.dispatchEvent(event);"
+                );
+            } catch (Exception $e) {
+                echo 'Caught exception: ',  $e->getMessage(), "\n";
+                        $attempts++;
+                        sleep(1);
+                        continue;
+            }
+            break;
+        } while($attempts < $this->NUM_OF_ATTEMPTS);
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
        } while($attempts < self::MAX_ATTEMPTS);
```

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 6]
```diff
@@ -535,5 +535,133 @@ abstract class LorisIntegrationTest extends TestCase
...
+                "
+                );
+            } catch (Exception $e) {
+                echo 'Caught exception: ',  $e->getMessage(), "\n";
+                        $attempts++;
+                        sleep(1);
+                        continue;
+            }
+            break;
+        } while($attempts < $this->NUM_OF_ATTEMPTS);
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
        } while($attempts < $this->self::MAX_ATTEMPTS);
```

File: test/integrationtests/LorisIntegrationTest.class.inc [hunk 7]
```diff
@@ -37,7 +37,10 @@ abstract class LorisIntegrationTest extends TestCase
     protected $factory;
 
     private $_oldConfig = array();
-
+    /** The maximum number of attempts the integration tests will try
+     * to execute some task before aborting.
+     */
+    private const $NUM_OF_ATTEMPTS = 5;
```
Comment (johnsaigle) [inline test/integrationtests/LorisIntegrationTest.class.inc]:
```suggestion
    private const NUM_OF_ATTEMPTS = 5;
```

The `$` is not needed for constants

**Comment by kongtiaowang** [comment]
replace by https://github.com/aces/Loris/pull/4939

**Pull request closed without merging**